pkg/model/deployment_test.go

```diff
@@ -169,6 +170,35 @@ spec:
 	assert.Equal(t, "my-vol", model.backstageDeployment.deployment.Spec.Template.Spec.Volumes[3].Name)
 }
 
+func TestImageInCRPrevailsOnEnvVar(t *testing.T) {
+	bs := *deploymentTestBackstage.DeepCopy()
+	bs.Spec.Deployment = &bsv1.BackstageDeployment{}
+	bs.Spec.Deployment.Patch = &apiextensionsv1.JSON{
```

Comment:
IMO, this test makes sense but I am missing a test that mimics the repro steps depicted in https://issues.redhat.com/browse/RHIDP-4013, i.e., using `bs.Spec.Application.Image` instead.
Here the conditions are different to me, and do not guarantee that the exact issue reported will not occur again.

Could we change this test (or add a separate test) using `bs.Spec.Application.Image` instead? And also make sure that both the init and main containers are set using this field (this is the current behavior expected from the `spec.application.image` field, which I recall was tested in the past before being removed at some point).

Reply:
This test the issue we had I believe, the problem was not in bs.Spec.Application.Image, it was env var vs CR config.
So, I'd avoid to make extra (duplicating) tests for deprecated functionality

Reply:
The issue was explicitly with `spec.application.image`, not with `spec.deployment` :)
And in that same file, you already have a test for "deprecated functionality", with an explicit comment `to remove when stop supporting v1alpha1`: https://github.com/redhat-developer/rhdh-operator/blob/f69ef79ea8c303e05e9c83c4d0ee14add9f32a2c/pkg/model/deployment_test.go#L202-L229
That's why I was thinking it would be safer to have it tested.
(Also, deprecated should not mean no longer tested/supported ^^)

Reply:
>The issue was explicitly with spec.application.image, not with spec.deployment :)

yes, it was but the root reason (unless you found something else) was not about spec.application.image

> And in that same file, you already have a test for "deprecated functionality", with an explicit comment to remove when stop supporting v1alpha1

yes, and I consider it would be enough. 

> Also, deprecated should not mean no longer tested/supported ^^)

yes, and it is tested and supported, see for example (as minimum)
Unit tests:
deployment_test.go
- TestSpecs
- TestDeploymentFieldPrevailsOnDeprecated

Integration tests:
- cr-config_test.go

e2e with 
rhdh-cr.yaml
rhdh-cr-with-app-configs.yaml

IMO it is quite descent set for 3 fields.